kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/Either.java

```diff
@@ -0,0 +1,35 @@
+package org.kie.dmn.feel.util;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+public class Either<L,R> {
```

Comment:
big thanks to @mariofusco for reminding me this was the perfect use case to apply FP-Either